scripts/GenerateTags.ps1

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env pwsh
+
+<#
+.SYNOPSIS
+    Generates a tag on this repo and adds a tag for each submodule that corresponds
+    to the value in version.props
+.PARAMETER Push
+    Push the tag to origin
+.PARAMETER WhatIf
+    Dry run
+#>
+[cmdletbinding(PositionalBinding = $false, SupportsShouldProcess = $true)]
+param(
+    [switch]$Push
+)
+
+$ErrorActionPreference = 'Stop'
+Import-Module -Scope Local -Force "$PSScriptRoot/common.psm1"
+Set-StrictMode -Version 1
+
+function New-GitTag {
+    [cmdletbinding(SupportsShouldProcess = $true)]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Repo,
+        [Parameter(Mandatory = $true)]
+        [string]$Tag
+    )
+
+    Push-Location $Repo
+    try {
+        $tags = & git show-ref --tags
+        $existingTag = $tags | where-object { $_ -like "*refs/tags/$Tag" }
+
+        if ($existingTag) {
+            Write-Warning "${Repo}: Tag '$Tag' already exists. Skipped adding tag"
+        }
+        else {
+            if ($PSCmdlet.ShouldProcess($Repo, "Tag $Tag")) {
+                Invoke-Block { & git tag -m "v$Tag" $Tag HEAD }
+                Write-Host -f Magenta "${Repo}: added tag '$Tag'"
+            }
+        }
+
+        if ($Push -and $PSCmdlet.ShouldProcess($Repo, "Push tag $Tag to origin")) {
+            Invoke-Block { & git push origin refs/tags/$Tag }
+        }
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../"
+
+Write-Warning "Make sure you have run ``git submodule update`` first to pin the submodules to the correct"
+
+Push-Location $repoRoot
+try {
+    [xml] $universeVersion = Get-Content "version.props"
```

Comment:
Can we add a KoreBuild target that spits out the version instead? It doesn't feel right to duplicate our version construction behavior in ps